[Ivander] iP

ivanderjmw opened the pull request.

ivanderjmw committed:
Add Greeting

Level-0 increment. Create a skeletal version that simply greets the user and exists.

ivanderjmw committed:
Echo user input

Level-1. Greets, Echo, Exit.
Add printOutputFormat() function.

ivanderjmw committed:
Create tasks feature, add and list methods
Level-2. Add the ability to store whatever text the user requested.

ivanderjmw committed:
Add mark as done feature.
Level-3. Enables users to set a task to done.
Add error handling for indices larger than task count.

ivanderjmw committed:
Modified to follow the Coding Standard.
A-CodingStandard.
Renamed some variables to fit the coding standard naming convention.
Add some comments.

johanesrafael commented on src/main/java/Duke.java:
```diff
@@ -5,11 +5,17 @@
     private static Task[] tasks;
     private static int taskCount;
 
-    public static void printOutputFormat(String output) {
```
Hey I like how you created a method for the output format!

sixletters replied:
Same!

johanesrafael reviewed (commented):
Looks good 👍

ivanderjmw committed:
Create Todos, Events, and Deadlines with Inheritance

sixletters commented on src/main/java/Duke.java:
```diff
@@ -24,29 +30,28 @@ public static void printWelcomeMessage() {
 
     public static void addTask(String task) {
 
         tasks[taskCount] = new Task(task);
```
Instead of using arrays and counting the number of tasks. You could also use arraylists from java docs to access more functionalities

ivanderjmw replied:
That is a good suggestion! Thank you, I will try to read up on it.

ivanderjmw committed:
Refine todos, events, deadlines, and parsing functions.
Level-4.

ivanderjmw committed:
Add automated text UI testing.
A-TextUiTesting.

ivanderjmw committed:
Improve code quality
A-CodeQuality

ivanderjmw committed:
Handle Errors with print statements.
Level-5.

ivanderjmw committed:
Use exceptions to handle errors.
A-Exceptions. Create an exception class Duke Exception.java.

ivanderjmw committed:
Include exceptions in UI testing.

ivanderjmw committed:
Merge branch 'branch-level-5'

ivanderjmw committed:
Refactor into packages.
A-Packages.

ivanderjmw committed:
Update README.md

ivanderjmw committed:
Update runtest scripts

ivanderjmw committed:
Merge branch 'branch-A-Packages'

ivanderjmw committed:
Implement Delete and use arraylist.
Level-6 and A-Collections.

ivanderjmw committed:
Create filesystem storage.
Level-7.
Automatically saves tasks into a txt file (store.txt).
Loads an existing file during startup, else creates a new file.

ivanderjmw committed:
Create filesystem storage.
Level-7.
Automatically saves tasks into a txt file (store.txt).
Loads an existing file during startup, else creates a new file.
Uses OS-independent file paths.

ivanderjmw committed:
Merge branch 'branch-Level-6'

ivanderjmw committed:
Merge Delete and Save feature.
Level-6. Level-7.

ivanderjmw committed:
Fix deferred absolute path.
Added a new relative path and `duke_data` folder to store tasks.
Fixed some redundant error handling.

Tejasvas123 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,302 @@
+package duke;
+
+import duke.task.*;
```
Following java coding standards, `*` should be replaced with individual import statements

Tejasvas123 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,302 @@
+
+                /*
+                  type | isDone | description | attribute
+                  The input array splits the encrypted input into separate String values
+                 */
+
+                switch (inputArray[0]) {
+                case "T":
+                    t = new ToDo(inputArray[2]);
+                    break;
+                case "D":
+                    t = new Deadline(inputArray[2], inputArray[3]);
+                    break;
+                case "E":
+                    t = new Event(inputArray[2], inputArray[3]);
+                    break;
+                default:
+                    throw new DukeException("Parse Error");
+                }
+
+                if (inputArray[1].equals("1")) {
+                    t.setDone();
+                }
+
+                parsedTasks.add(parsedTaskCount, t);
+                parsedTaskCount++;
+            }
+
+            tasks = parsedTasks;
+            taskCount = parsedTaskCount;
+
+        } catch (FileNotFoundException e) {
+            try {
+                createFile();
+            } catch (IOException k) {
+                throw new DukeException(k.toString() + "\nProblem with reading the storage file.");
+            }
+        }
+
+    }
```
Try to refactor this method into bunch of smaller methods (as a general rule no method should be more than 30 lines)

Tejasvas123 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,302 @@
+
+    }
+
+    public static void parseAndAddTask(String taskData) throws DukeException {
```
All public methods of a class must be provided with a descriptive header comment

Tejasvas123 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,302 @@
+            }
+
+            String at = taskData.substring(taskData.indexOf("/at") + "/at".length()).trim();
+            String description = taskData.substring(0, taskData.indexOf("/at"))
```
Try to break compicated expressions into smaller simpler expressions

Tejasvas123 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,302 @@
+
+        if (taskIndex > taskCount - 1) {
+            throw new DukeException(
+                    "duke.task.Task number " + (taskIndex + 1)
```
String literals can be made named constants instead.

Tejasvas123 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,302 @@
+
+
+        switch (commandEntered) {
+        case "todo": case "deadline": case "event":
```
split into multiple lines

Tejasvas123 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,302 @@
+        case "done":
+            setTaskDone(rawInput);
+            break;
+        case "delete":
```
string literals like `delete`, `list` etc can be defined as constants instead

Tejasvas123 reviewed (commented):
Good job! Remember to add comments to your code!

ivanderjmw committed:
Implement more OOP #1

ivanderjmw committed:
Add more OOP for commands

ivanderjmw committed:
Add find command

ivanderjmw committed:
Add JavaDoc comments

ivanderjmw committed:
Update unused imports

ivanderjmw committed:
Merge pull request #2 from ivanderjmw/branch-Level-9

Add find command

ivanderjmw committed:
Merge branch 'master' into branch-JavaDoc

ivanderjmw committed:
Merge pull request #3 from ivanderjmw/branch-JavaDoc

Add JavaDoc

ivanderjmw committed:
Fix null command

ivanderjmw committed:
Add user guide

ivanderjmw committed:
Modify user guide

ivanderjmw committed:
Set theme jekyll-theme-slate

ivanderjmw committed:
Set theme jekyll-theme-minimal

ivanderjmw committed:
modify output in list command

ivanderjmw committed:
Merge branch 'master' of github.com:ivanderjmw/ip

ivanderjmw committed:
Modify user guide

ivanderjmw committed:
Fix markdown bug

ivanderjmw committed:
Create gradle.yml

ivanderjmw committed:
Update README.md

ivanderjmw committed:
Update README.md

ivanderjmw committed:
Fix character encoding and storage bug.

ivanderjmw committed:
Add javadoc comments to functions.
